### PR TITLE
Initial release.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

**Comment by lask** [comment]
Add a `src/.gitkeep`? I would prefer that there was a `src` folder by default.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,73 @@
...
+Examples and tests can have different dependencies than the package. This is,
+why they have their own package.yaml/package.lock.
+
+Open the examples (resp. tests) folder with a separate instance of your IDE.
+For vscode you could just write `code examples`.
+
+Install this package as a local package.
+```
+cd examples
+toit.pkg init
```
Comment (lask) [inline README.md]: Maybe initialize the package as part of the template? I find it unlikely that you would ever have an examples folder in a package without needing the package.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,73 @@
...
+that the first paragraph of this README can be used as a description.
+
+## Copyright
+Don't forget to update the copyright holder in the license files.
+There are (up to) three license files:
+- `LICENSE`: usually MIT
+- `examples/EXAMPLES_LICENSE`: usually BSD0
+- `tests/TESTS_LICENSE`: usually BSD0
+
+We recommend to use the following Copyright header in `src` files (with your
```
Comment (lask) [inline README.md:31]: Maybe mention that the package code goes in the `src` folder and that that folder needs to be added. As I mentioned in a comment, I would prefer to have the `src` folder around as part of the template.

**Review by lask — APPROVED** [review]
Looks good! I would prefer to add as much boilerplate as possible.

File: README.md (hunk 2 above)
Comment (floitsch) [reply README.md:31]: done.

File: README.md (hunk 1 above)
Comment (floitsch) [reply README.md]: done.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,81 @@
-# template-pkg
-A template repository for creating a Toit package
+# Package template
+
+A template repository for creating a Toit package.
+
+## Toit package
+Use `toit.pkg describe` or `toit pkg describe` (depending on which Toit
+variant you use) to see how https://pkg.toit.io extracts package
+information.
```
Comment (erikcorry) [inline README.md:8]: Suggestion: "to see how https://pkg.toit.io will extract package information
from your repo when you publish your package."

File: README.md [hunk 4]
```diff
@@ -1,2 +1,81 @@
-# template-pkg
-A template repository for creating a Toit package
+# Package template
```
Comment (erikcorry) [inline README.md:1]: Suggestion: Call the package Mypackage.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,81 @@
...
+There are (up to) three license files:
+- `LICENSE`: usually MIT
+- `examples/EXAMPLES_LICENSE`: usually BSD0
+- `tests/TESTS_LICENSE`: usually BSD0
+
+We recommend to use the following Copyright header in `src` files (with your
+copyright):
+
+```
+// Copyright (C) 2022 Toit language
```
Comment (erikcorry) [inline README.md:35]: Suggestion: Use "Copyright (C) 2022 Jo Doe"

File: README.md [hunk 6]
```diff
@@ -1,2 +1,81 @@
...
+Install this package as a local package.
+```
+cd examples
+toit.pkg install --local --name=YOUR_PACKAGE_NAME ..
+```
+
+This allows examples and tests to import this package as if it was installed
+from the Internet. This way, tests and examples use the same syntax as
+users of the package.
+
```
Comment (erikcorry) [inline README.md:69]: Suggestion: Mention using .. as the path of the parent package?

Comment (erikcorry) [reply README.md:69]: and show it in the package.yaml/lock files?

File: README.md [hunk 7]
```diff
@@ -1,2 +1,81 @@
...
+A template repository for creating a Toit package.
+
+## Toit package
+Use `toit.pkg describe` or `toit pkg describe` (depending on which Toit
+variant you use) to see how https://pkg.toit.io extracts package
+information.
+
+Either add a `name: ...` entry to the package.yaml or change the title
+of this README to the package name.
```
Comment (erikcorry) [inline README.md:11]: Suggestion: change the title (first line) of this README

File: README.md (hunk 4 above)
Comment (floitsch) [reply README.md:1]: done.

File: README.md (hunk 3 above)
Comment (floitsch) [reply README.md:8]: done.

File: README.md (hunk 7 above)
Comment (floitsch) [reply README.md:11]: done.

File: README.md (hunk 5 above)
Comment (floitsch) [reply README.md:35]: Jane/John Doe.
done.

File: README.md (hunk 6 above)
Comment (floitsch) [reply README.md:69]: reworded,
but not mentioning the package.yaml/lock.

**Review by floitsch — COMMENTED** [review]
Addressed in https://github.com/toitlang/template-pkg/pull/2

**Pull request merged**